src/components/Camera.tsx

```diff
@@ -18,25 +20,41 @@ export const Camera = () => {
 
     let stream: MediaStream | null = null;
     let isTerminated = false;
+    let intervalId: NodeJS.Timeout | null = null;
 
     const startStream = async () => {
       try {
-        stream = await navigator.mediaDevices.getUserMedia({
-          video: true,
-        });
+        stream = await navigator.mediaDevices.getUserMedia({ video: true });
 
-        if (isTerminated) {
+        if (isTerminated || !videoRef.current) {
           stream.getTracks().forEach((track) => track.stop());
           return;
         }
 
-        videoRef.current!.srcObject = stream;
+        videoRef.current.srcObject = stream;
+
+        const model = handPoseDetection.SupportedModels.MediaPipeHands;
+        const detectorConfig: handPoseDetection.MediaPipeHandsMediaPipeModelConfig =
+          {
+            runtime: "mediapipe",
+            solutionPath:
+              "https://cdn.jsdelivr.net/npm/@mediapipe/hands@0.4.1646424915/",
+            modelType: "full",
+          };
+
+        const detector = await handPoseDetection.createDetector(
```

Comment:
After this await isTerminated can also be true. I recommend:

Create function `clenup` as it was before: stream.getTracks().forEach((track) => track.stop());
Add detector.dispose() to this cleanup function because it is required
Call it after awaits if the hook was terminated
Call it in the useEffect cleanup flow